pulser-core/pulser/json/abstract_repr/serializer.py

```diff
@@ -53,6 +61,11 @@ def default(self, o: Any) -> dict[str, Any] | list | int | float:
                 # Try to return a real number when possible
                 return o.real
             return dict(real=o.real, imag=o.imag)
+        elif has_torch and isinstance(o, torch.Tensor):
+            if len(o.shape) == 0:
+                return o.item()
+            else:
+                return o.tolist()
```

Comment:
This is not needed, `tolist()` on a 0D tensor returns a scalar
```suggestion
            return o.tolist()
```

Reply:
I see that indeed it does. But that is hideously misleading.